share/scripts/sdcdb.tcl

```diff
@@ -490,22 +491,23 @@ proc list_file {file begin {end {}} {focus 0} {showerror 0}} {
         if {$showerror} {
             error "file '$file' not found in database, add a directory that contains such file with 'sdcdb add <dir>'"
         }
-        puts "$file: not found"
-        return
+        return "$file: not found"
     }
     debug_out "opening file [lindex $record 1]..."
     set fh [open [lindex $record 1] r]
     set pos 0
     # 10 lines by default
     if {$end eq {}} { set end [expr {$begin + 9}] }
+    set output [list]
     while {[gets $fh line] >= 0} {
         incr pos
         if {$pos >= $begin} {
-            puts "[format %-5d $pos][expr {$pos == $focus ? "*" : ":"}]    $line"
+            lappend output "[format %-5d $pos][expr {$pos == $focus ? "*" : ":"}]    $line"
         }
         if {$pos >= $end} { break }
     }
     close $fh
+    join $output "\n"
```

Comment:
This is also fine (creating a list, and then in the end join that list with `\n`).
Above I used `append output "...\n"`.
Pick which you prefer best.